lib/Sema/BuilderTransform.cpp

```diff
@@ -699,6 +699,18 @@ class BuilderClosureVisitor
     if (!cs)
       return nullptr;
 
+    // If there are no 'case' statements in the body let's try
+    // to diagnose this situation via limited exhaustiveness check
+    // before failing a builder transform, otherwise type-checker
+    // might end up without any diagnostics which leads to crashes
+    // in SILGen.
+    if (capturedCaseVars.empty()) {
+      TypeChecker::checkSwitchExhaustiveness(switchStmt, dc,
+                                             /*limitChecking=*/true);
```

Comment:
We check for exhaustiveness in `BuilderClosureRewriter` as well, is that unreachable in this scenario?

Reply:
Yes, in the case where there aren't any valid expressions in the switch, the solver won't find a solution and won't continue to the rewrite stage

Reply:
Okay, thank you for the explanation!

Reply:
Though we just figure out that this change still needs some work because it will still crash (in a different way) if there is a case term but no expression